examples/knative/python/kn-py-vro/handler.py

```diff
@@ -0,0 +1,270 @@
+        param['value'] = {"number": {"value": value}}
+
+    elif isinstance(value, str):
+        try:  # for strings, try and parse to a date first...
+            isoparse(value)
+            param['type'] = "Date"
+            param['value'] = {"date": {"value": value}}
+        except ValueError:  # ...if that doesn't work just build a string
+            param['type'] = "string"
+            param['value'] = {"string": {"value": value}}
+
+    elif type(value) == bool:
+        param['type'] = "boolean"
+        param['value'] = {"boolean": {"value": value}}
+
+    elif type(value) == dict:  # a dict is likely a VC manged object reference
+        for v in value.values():
+            # search the items of the dict and look for contained dicts with a 'Type' property
+            if type(v) == dict and 'Type' in v:
+                param['type'] = f"VC:{v['Type']}"
+                param['value'] = {
+                    "sdk-object": {
+                        "type": param['type'],
+                        "id": f'{vchost},id:{v["Value"]}'
+                    }
+                }
+        if not 'type' in param: # if no vc managed object was found, pass the data as JSON
+            param['type'] = 'string'
+            param['value'] = {"string": {"value": json.dumps(value)}}
+
+    elif type(value) == list:
+        param['type'] = "Properties"
+        properties = []
+        for v in value:
+            if 'Key' in v and 'Value' in v:
+                # pass the key and value to this function recursively to get a correctly structured value
+                vroParam = getVroInputParam((v["Key"], v["Value"]))
+                properties.append({"key": v["Key"], "value": vroParam["value"]})
+        param['value'] = {"properties": {"property": properties}}
+
+    elif value is None:
+        param = None
+
+    else:  # Unhandled data type - try and wrangle to a string
+        app.logger.debug(f'{bgc.WARNING}Unhandled data type "{type(value).__name__}" - forcing to string{bgc.ENDC}')
+        param['type'] = "string"
+        param['value'] = {"string": {"value": str(value)}}
+
+    if type(param) == dict:  # set the parameter name for valid types
+        param["name"] = name[0].lower() + name[1:]
+    return param
+
+
+def get_response(message, code):
+        app.logger.error(message)
+        msg = {
+            'status': code,
+            'error': message,
+        }
+        resp = jsonify(msg)
+        resp.status_code = code
+        return resp
+
+
+app = Flask(__name__)
+@app.route('/', methods=['POST'])
+def handle():
+    """
+    Handles a request to the function
+
+    """
+    global vchost
+
+    # Load the Events that function gets from vCenter through the Event Router
+    app.logger.debug(f'{bgc.HEADER}Reading Cloud Event: {bgc.ENDC}')
+    cevent = from_http(request.headers, request.get_data(),None)
+
+    if cevent._attributes.get("datacontenttype").lower() != "application/json":
```

Comment:
nit: I think for CE attributes you can directly use dict syntax and not internal methods